lai_s3/s3.py

```diff
@@ -117,18 +125,82 @@ def _upload_file(
             self,
             bucket: str,
             object: str,
-            filename: Union[L.storage.Path, str]
+            filename: Union[lapp.storage.Path, str]
     ):
         with open(filename, 'rb') as _f:
             self.resource.meta.client.upload_fileobj(
                 Fileobj=_f, Bucket=bucket, Key=object
             )
 
     def run(self, action, *args, **kwargs):
-
         if action == "get_filelist":
             self._get_filelist(*args, **kwargs)
         elif action == "download_file":
             self._download_file(*args, **kwargs)
         elif action == "upload_file":
             self._upload_file(*args, **kwargs)
+        elif action == 'create_dataset':
+            return self._create_dataset(*args, **kwargs)
+
+
+    def _get_s3_items(data, idx, transforms=None, data_type='img', label_map=None):
+        if data_type.lower() == 'img':
+            obj = data[idx]
+            label = obj.key.split('/')[-2]
+            label = label_map[label]
+            img_bytes = obj.get()['Body'].read()
+            img = Image.open(io.BytesIO(img_bytes)).convert('RGB')
+            out = [img, label]
+        elif data_type.lower() == 'tabular':
```

Comment:
I could be missing something but I don't think this uses s3.  It looks like a dataframe is passed is passed into the function.  Generally, not a great idea to do features/label based on column ordering.